server/controllers/challenges.js

```diff
@@ -112,14 +110,17 @@ const challengesController = {
             return res.status(200).send({leafNodeTransactions});
         }
 
-        allTransactions.forEach(async transaction => {
+        allTransactions.forEach(async (transaction, index) => {
```

Comment:
so for this function 'retrieve all leaf node transactions'...can't you just query any of the transactions that have no children.

or is the only link from child to parent?

Reply:
fwiw in the real api/core it is child to parent -- however we could have two way link in this shitbox lol